APIs_RTOS/ThisThread_Blinky/main.cpp

```diff
@@ -0,0 +1,24 @@
+#include "mbed.h"
+#include "rtos.h"
+
+#define STOP_FLAG 1
+
+// Blink function toggles the LED in a long running loop
+void blink(DigitalOut *led)
+{
+    while (!ThisThread::flags_wait_any_for(STOP_FLAG, 1s)) {
+        *led = !*led;
+    }
+}
+
+// Spawns a thread to run blink for 5 seconds
```

Comment:
I am not sure I understand what all the comments in this code are telling me...Can we make improve the comment? I cannot even provide a solution as I do not understand what is the information conveyed.

Reply:
I think what they are trying to say is that the application creates a thread wherein an LED is toggled periodically, this thread is destroyed after 5 seconds.

Reply:
Probably sufficient to say something like this: 
```// Creates thread to toggle an LED for 5 seconds ```

Reply:
if that's the intended meaning, can you please update it?

Reply:
How about the other comment in the snippet?